Update analyzer docs (#223)
replicatedhq/troubleshoot.sh: docs/source/analyze/image-pull-secrets.md

description: Analyzing the image pull secrets that are available in the cluster
---

The `imagePullSecrets` analyzer uses data from the [clusterResources collector](https://troubleshoot.sh/collect/cluster-resources).
The `clusterResources` collector is automatically added and will always be present.

## Parameters

**registryName**: (Required) The name of the registry to check

```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: Preflight
metadata:
  name: preflight-sample
spec:
  analyzers:
    - imagePullSecret:
        checkName: Pull from Quay
        registryName: quay.io
        outcomes:
          - fail:
              message: You won't be able to pull from Quay
          - pass:

>               message: Images can be pulled from Quay

Reviewer: Does this actually test the ability to pull from quay.io, or just test that there's a secret for that registry? The secret could be invalid.

Author: Only that a secret exists for that registry.

description: Analyzing the image pull secrets that are available in the cluster
---

The `imagePullSecret` analyzer checks that a secret exists with credentials to pull images from a registry.
It does not verify that the credentials in the secret are valid.

The `imagePullSecret` analyzer uses data from the [clusterResources collector](https://troubleshoot.sh/collect/cluster-resources).
The `clusterResources` collector is automatically added and will always be present.

## Parameters

**registryName**: (Required) The name of the registry to check

```yaml
apiVersion: troubleshoot.sh/v1beta2
kind: Preflight
metadata:
  name: preflight-sample
spec:
  analyzers:
    - imagePullSecret:
        checkName: Pull from Quay
        registryName: quay.io
        outcomes:
          - fail:

>               message: You won't be able to pull from Quay

Reviewer: ```suggestion
              message: Did not find credentials to pull from Quay
```

          - pass:

>               message: Images can probably be pulled from Quay

Reviewer: ```suggestion
              message: Found credentials to pull from Quay
```
